dlme_airflow/task_groups/etl.py

```diff
@@ -38,18 +37,32 @@ def build_provider_etl_taskgroup(provider, dag: DAG) -> TaskGroup:
     return provider_etl_taskgroup
 
 
-def build_collection_etl_taskgroup(provider, collection, post_harvest, task_group: TaskGroup, dag: DAG) -> TaskGroup:
+def build_collection_etl_taskgroup(provider, collection, task_group: TaskGroup, dag: DAG) -> TaskGroup:
+    catalog_key = provider
+    if collection:
+       catalog_key = f"{catalog_key}.{collection}"
+    
+    # Move fetching of post harvest from metadata into the task group
+    source = catalog_for_provider(catalog_key)
+    post_harvest = source.metadata.get('post_harvest', None)
+
+    logging.info(f"POST HARVEST={post_harvest}")
     with TaskGroup(
         group_id=f"{collection}_etl",
         dag=dag) as collection_etl_taskgroup:
             harvest = build_havester_task(provider, collection, collection_etl_taskgroup, dag)  # Harvest
-            post_harvest = build_post_havest_task(provider, collection, post_harvest, collection_etl_taskgroup, dag) # Post Harvest
             sync = build_sync_metadata_taskgroup(provider, collection, dag)
             transform = build_transform_task(provider, collection, collection, collection_etl_taskgroup, dag)  # Transform
             load = index_task(provider, collection, collection_etl_taskgroup, dag)  # Load / Index
             report = build_harvest_report_task(provider, collection, collection_etl_taskgroup, dag)  # Report
             send_report = build_send_harvest_report_task(provider, collection, collection_etl_taskgroup, dag)  # Send Report
 
-            harvest >> post_harvest >> sync >> transform >> load >> report >> send_report
+            # If we fetch a post_harvest key from the catalog, build the post_harvest_task and include it in the flow
+            if post_harvest:
```

Comment:
I moved this further down inside of a conditional so that we avoid adding unnecessary post_harvest steps.